### PR TITLE
Allow user to provide ocean resolution at setup_expt invocation

### PR DESCRIPTION
**Pull request by aerorahul**
# Description
This PR:
- allows a user to provide a desired ocean resolution during the invocation of `setup_expt.py`.
- renames `--resdet` option to `--resdetatmos`
- adds `--resdetocean`.  If a value `<= 0.` is provided, the program will choose based on pre-selected options.
- updates the CI test yamls
- updates the documentation

# Type of change
- New feature (adds functionality)
- Maintenance (code refactor)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? No, but users will need to update their use of `setup_expt.py`
- Does this change require a documentation update? YES

# How has this been tested?
- Created a few of the experiment directories using `create_experiment.py` and the various `ci/cases` yaml files and inspected `config.base` in the respective experiment directories.

# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

File: docs/doxygen/mainpage.h [hunk 1]
```diff
@@ -40,9 +40,11 @@ To setup an experiment, a python script <b>\c setup_expt.py</b> (located in <b>\
                         (default: 2016100100)
         --icsdir     full path to initial condition directory
                         (default: /scratch4/NCEPDEV/da/noscrub/Rahul.Mahajan/ICS)
-        --resdet     resolution of the deterministic model forecast
+        --resdetatmos atmosphere resolution of the deterministic model forecast
                         (default: 384)
-        --resens     resolution of the ensemble model forecast
+        --resdetocean ocean resolution of the deterministic model forecast
+                        (default: 0. [determined automatically based on atmosphere resolution]]])
```
Comment (KateFriedman-NOAA) [inline docs/doxygen/mainpage.h]: Do you mean to have three closing brackets on line 46?

File: workflow/setup_expt.py [hunk 2]
```diff
@@ -521,13 +524,32 @@ def validate_user_request(host, inputs):
...
 
+def get_ocean_resolution(resdetatmos):
+    """
+    Method to determine the ocean resolution based on the atmosphere resolution
+    Limited options are going to be available
+    """
+    atmos_to_ocean_map = {
+        1152: 0.25, 768: 0.25, 384: 0.25,
+        192: 1.0,
+        96: 5.0, 48: 5.0}
```
Comment (KateFriedman-NOAA) [inline workflow/setup_expt.py:535]: Should this table be added to RTD?

**Review by KateFriedman-NOAA — APPROVED** [review]
Looks good! Left a few comments but approve pending any CI testing.

File: parm/config/gefs/config.base.emc.dyn [hunk 3]
```diff
@@ -154,35 +154,27 @@ export OPS_RES="C768" # Do not change
...
 export CASE_ENS="@CASEENS@"
-# TODO: This should not depend on $CASE or $CASE_ENS
-# These are the currently available grid-combinations
+export OCNRES="@OCNRES@"
+export ICERES="${OCNRES}"
+# These are the currently recommended grid-combinations
 case "${CASE}" in
     "C48")
-        export OCNRES=500
         export waveGRD='glo_500'
```
Comment (WalterKolczynski-NOAA) [inline parm/config/gefs/config.base.emc.dyn:162]: If we're decoupling the marine resolution from the atmosphere resolution, should we also decouple the wave grid?

File: workflow/setup_expt.py [hunk 4]
```diff
@@ -521,13 +524,32 @@ def validate_user_request(host, inputs):
             raise NotImplementedError(f"Supported resolutions on {machine} are:\n{', '.join(supp_res)}")
 
 
+def get_ocean_resolution(resdetatmos):
```
Comment (WalterKolczynski-NOAA) [inline workflow/setup_expt.py:527]:
```suggestion
def get_default_ocean_resolution(resdetatmos):
```

File: parm/config/gefs/config.base.emc.dyn (hunk 3 above)
Comment (aerorahul) [reply parm/config/gefs/config.base.emc.dyn:162]: some other time.

File: docs/doxygen/mainpage.h (hunk 1 above)
Comment (aerorahul) [inline docs/doxygen/mainpage.h]:
```suggestion
                        (default: 0. [determined automatically based on atmosphere resolution])
```

File: workflow/setup_expt.py (hunk 2 above)
Comment (aerorahul) [reply workflow/setup_expt.py:535]: I couldn't find a spot for it so didn't.

File: docs/doxygen/mainpage.h (hunk 1 above)
Comment (aerorahul) [reply docs/doxygen/mainpage.h]: Are we even building doxygen based documentation anymore?

File: workflow/setup_expt.py (hunk 2 above)
Comment (KateFriedman-NOAA) [reply workflow/setup_expt.py:535]: The initial conditions page has the atmosphere and ocean resolutions we support but perhaps this info should be on a different page now that I look at it: https://global-workflow.readthedocs.io/en/latest/init.html

File: ci/cases/pr/C96_atm3DVar.yaml [hunk 5]
```diff
@@ -5,10 +5,10 @@ experiment:
 arguments:
   pslot: {{ 'pslot' | getenv }}
   app: ATM
-  resdet: 96
+  resdetatmos: 96
   comrot: {{ 'RUNTESTS' | getenv }}/COMROT
   expdir: {{ 'RUNTESTS' | getenv }}/EXPDIR
-  icsdir: ${ICSDIR_ROOT}/C96C48
+  icsdir: {{ 'ICSDIR_ROOT' | getenv ]}/C96C48
```
Comment (aerorahul) [inline ci/cases/pr/C96_atm3DVar.yaml]:
```suggestion
  icsdir: {{ 'ICSDIR_ROOT' | getenv }}/C96C48
```

**Comment by aerorahul** [comment]
Absorbed in #2204.
Closing.

**Pull request closed without merging**